fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -578,11 +580,19 @@ private void disburseLoan(JsonCommand command, boolean isPaymentTypeApplicableFo
         final Money interestApplied = Money.of(loan.getCurrency(), loan.getSummary().getTotalInterestCharged());
 
         /*
-         * Add an interest applied transaction of the interest is accrued upfront (Up front accrual), no accounting or
-         * cash based accounting is selected
+         * Add an interest applied transaction if the interest is accrued upfront (Up front accrual)
          */
-        if (((loan.isMultiDisburmentLoan() && loan.getDisbursedLoanDisbursementDetails().size() == 1) || !loan.isMultiDisburmentLoan())
-                && loan.isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct() && interestApplied.isGreaterThanZero()) {
+
+        boolean isSingleDisbursement = loan.isMultiDisburmentLoan() && loan.getDisbursedLoanDisbursementDetails().size() == 1;
```

Comment:
This is pretty misleading naming... would you consider renaming to `isFirstDisbursement` and the value is:
`(loan.isMultiDisburmentLoan() && loan.getDisbursedLoanDisbursementDetails().size() == 1) || !loan.isMultiDisburmentLoan())`

This way is more straightforward... what it is for...